docs/db_connection_security.rst

```diff
@@ -0,0 +1,12 @@
+Database Connection Security
```

Comment:
@cotedm for the include, the convention is to nest in an `/includes` folder (e.g. `connect-jdbc/includes/db_connection_security.rst`). also, not an issue here, but eventually if you have more than one string to include, you can do this by referencing the specific line numbers for your include (see https://github.com/confluentinc/docs#content-reuse).

Reply:
i've added more explanation here as well https://github.com/confluentinc/docs/pull/650

Reply:
thanks @joel-hamill! how's this look? If it's good, I think we're ok to merge :)